### PR TITLE
Use the tpm_ownerpassword when generating the EK

### PR DESCRIPTION
**Pull request by ansasaki**
Set the `tpm_ownerpassword` in the TPM context to be used for the Endorsement Hierarchy regardless of the value set in `ek_handle`.

This makes the `tpm_ownerpassword` set in the configuration file to be used to access the TPM when generating the EK handle.

Note: This doesn't change the TPM owner password, only provide it when creating a password session.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -448,17 +448,14 @@ async fn main() -> Result<()> {
...
-            ctx.tr_set_auth(Hierarchy::Endorsement.into(), auth)?;
-        }
-    }
+    // When the tpm_ownerpassword is given, set auth for the Endorsement hierarchy.
+    // Note in the Python implementation, tpm_ownerpassword option is also used for claiming
+    // ownership of TPM access, which will not be implemented here.
+    if let Some(ref v) = config.tpm_ownerpassword {
+        let auth = Auth::try_from(v.as_bytes())?;
+        ctx.tr_set_auth(Hierarchy::Endorsement.into(), auth)
+            .expect("Failed to set auth for Endorsement Hierarchy");
```
Comment (ueno) [inline src/main.rs:457]: Wouldn't it be better to return error, rather than panic?

**Review by ueno — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (ansasaki) [reply src/main.rs:457]: I will change it to return error.

**Comment by lkatalin** [comment]
> Note: This doesn't change the TPM owner password, only provide it when creating a password session.

Thank you for the updated note.

**Review by lkatalin — APPROVED** [review]
LGTM since it does not set the password

**Pull request merged**